docs/knowledgebase/runtime/fees.md

```diff
@@ -1,113 +1,138 @@
-  WeightToFee(weight)
+inclusion_fee = base_fee + length_fee + [targeted_fee_adjustment * weight_fee];
+final_fee = inclusion_fee + tip;
 ```
 
-This `fee` is known as the "inclusion fee". Note that the extrinsic sender is charged the inclusion
-fee _prior_ to the actual invocation of the extrinsic, so its cost will still be incurred if
-execution fails. In the event that an account does not have a sufficient balance to pay the fee and
-remain alive (i.e. existential deposit plus inclusion fee), no fee will be deducted and the
-transaction will not begin execution. This latter case should be rare as the transaction queue and
-block construction logic perform checks prior to adding an extrinsic to a block.
+In this formula, the `targeted_fee_adjustment` is a multiplier that can tune the final fee based on the congestion of the network.
 
-### Fee Multiplier
+* The `base_fee` derived from the base weight covers inclusion overhead like signature verification.
 
-The above formula gives a fee that is always the same for the same input. However, weight can be
-dynamic and, based on how
-[`WeightToFee`](https://substrate.dev/rustdocs/latest/pallet_transaction_payment/pallet/trait.Config.html#associatedtype.WeightToFee)
-is defined, the final fee can include some degree of variability. To fulfill this requirement,
-Substrate provides:
+* The `length_fee` is a per-byte fee that is multiplied by the length of the encoded extrinsic.
 
-- [`NextFeeMultiplier`](https://substrate.dev/rustdocs/latest/pallet_transaction_payment/pallet/type.NextFeeMultiplier.html):
-  A configurable multiplier stored in the Transaction Payment pallet.
-- [`FeeMultiplierUpdate`](https://substrate.dev/rustdocs/latest/pallet_transaction_payment/pallet/trait.Config.html#associatedtype.FeeMultiplierUpdate):
-  A configurable parameter for a runtime to describe how this multiplier can change.
+* The `weight_fee` fee is calculated using two parameters:
 
-`NextFeeMultiplier` has the type `Fixed64`, which can represent a fixed point number. So, given the
-inclusion fee formula above, the final version would be:
+	 The `ExtrinsicBaseWeight` that is declared in the runtime and applies to all extrinsics. 
+	 
+	 The #[weight] annotation that accounts for an extrinsic's complexity.
 
-```
-fee =
-  len(tx) * length_fee +
-  WeightToFee(weight)
+To convert the weight to Currency, the runtime must define a `WeightToFee` struct that implements a conversion function, `Convert<Weight,Balance>`. 
 
-final_fee = fee * NextFeeMultiplier
-```
+Note that the extrinsic sender is charged the inclusion
+fee before the extrinsic is invoked. The fee is deducted from the sender's balance even if the transaction fails upon execution. 
+
+### Accounts with an insufficient balance
+
+If an account does not have a sufficient balance to pay the inclusion fee and remain alive—that is, enough to pay the inclusion fee and maintain the minimum **existential deposit**—then no fee is deducted and the
```

Comment:
I don't think the comment about remaining alive is true – some transactions make sure an account does not disappear as the result of them, but not all do. The latter can result in an account dropping below the existential deposit, and then disappearing.

Due to payments happening before execution, it is even possible this happens and then a refund is issued that would be enough to keep that account alive, but that cannot happen so the refund is just not issued at all – strange corner case that I saw handled in the code.

Reply:
The most obvious example is the difference between `transfer` (https://github.com/paritytech/substrate/blob/136120674703bf5541ee2368ebb330d803b61cce/frame/balances/src/lib.rs#L279) and `transfer_keep_alive` (https://github.com/paritytech/substrate/blob/136120674703bf5541ee2368ebb330d803b61cce/frame/balances/src/lib.rs#L390). The former might make an account go below the existence threshold, the latter ensures it doesn't. The system enforces nothing in this regard as far as I can tell.

On the other hand I'm actually not quite sure how this interacts with fees, since these are transfers and that's what ensuring continued existence means here. So, uh, what you wrote before might actually be right? <_<"